ui/jobs/combo_tracker.ts

```diff
@@ -27,11 +38,11 @@ export default class ComboTracker {
       if (!nextMap[id])
         nextMap[id] = node;
 
-      nextMap = nextMap[id].next;
-    }
+      nextMap = nextMap[id]?.next ?? {};
```

Comment:
I think this could be more readable without the unnecessary `?`.  `nextMap[id]` and `nextMap[id].next` are both always defined.  What about:

```typescript
      let nextEntry = nextMap[id];
      if (!nextEntry)
        nextEntry = nextMap[id] = node;
      nextMap = nextEntry.next;
```